src/Rx/Weno/TransmitProperties.php

```diff
@@ -0,0 +1,213 @@
+
+        if (empty($locid['weno_id'])) {
+            //if not in an encounter then get the first facility location id as default
+            $default_facility = sqlQuery("select name, street, city, state, postal_code, phone, fax, weno_id from facility order by id asc limit 1");
+
+            if (empty($default_facility)) {
+                echo xlt('Facility ID is missing');
+                exit;
+            } else {
+                return $default_facility;
+            }
+        }
+        return $locid;
+    }
+
+    /**
+     * @return mixed
+     */
+    private function getPatientInfo()
+    {
+        //get patient data
+        $patient = sqlQuery("select title, fname, lname, mname, street, state, city, email, phone_cell, postal_code, dob, sex, pid from patient_data where pid = ?", [isset($_SESSION['pid'])]);
+        if (empty($patient['fname']) || empty($patient['lname']) || empty($patient['dob']) || empty($patient['sex']) || empty($patient['postal_code']) || empty($patient['street']) || empty($patient['email'])) {
+            echo xlt('Patient data is incomplete phone or ')
+                . ", "
+                . xlt('first last name') . ", "
+                . xlt('gender email') . ", "
+                . xlt('zip code') . ", "
+                . xlt('date of birth or address');
+            exit;
+        }
+        return $patient;
+    }
+
+    /**
+     * @return string
+     * New Rx
+     */
+    public function cipherpayload()
+    {
+        $cipher = "aes-256-cbc"; // AES 256 CBC cipher
+        $cryptoGen = new Crypto\CryptoGen();
+        $enc_key = $cryptoGen->decryptStandard($GLOBALS['weno_encryption_key']);
+        if ($enc_key) {
+            $key = substr(hash('sha256', $enc_key, true), 0, 32);
+            $iv = chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0);
+            $ciphertext = base64_encode(openssl_encrypt($this->payload, $cipher, $key, OPENSSL_RAW_DATA, $iv));
+            return $ciphertext;
+        } else {
+            return "error";
+        }
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getProviderPassword()
+    {
+        $cryptoGen = new Crypto\CryptoGen();
+        $uid = $_SESSION['authUserID'];
+        $sql = "select setting_value from user_settings where setting_user = ? and setting_label = 'global:weno_provider_password'";
+        $prov_pass = sqlQuery($sql, [$uid]);
+        if ($prov_pass['setting_value']) {
+            return $cryptoGen->decryptStandard($prov_pass['setting_value']);
+        } else {
+            echo xlt('Password is missing');
+            die;
+        }
+    }
+
+    /**
+     * @return mixed
+     */
+    private function getVitals()
+    {
+        $vitals = sqlQuery("select date, height, weight from form_vitals where pid = ? ORDER BY id DESC", [isset($_SESSION["pid"])]);
```

Comment:
same issue as above with use of isset() function

Reply:
done